api/src/main/java/org/apache/iceberg/SnapshotUpdate.java

```diff
@@ -60,4 +60,9 @@
    * @return this for method chaining
    */
   ThisT scanManifestsWith(ExecutorService executorService);
+
+  /**
+   * Perform operations a particular branch
+   */
+  ThisT branch(String branch);
```

Comment:
I think we need to add another word to make it clear what is happening here, like `toBranch` or `forBranch`. Just branch isn't obvious when you're reading it:

```java
table.newOverwrite()
    .overwriteByRowFilter(filter)
    .addFile(one)
    .addFile(two)
    .toBranch(b)
    .commit()
```